File: 20-sortiranje/redjanje-umetanjem.py
def redjajUmetanjem(niz):
    for i in range(len(niz)):
        trenutni = niz[i]
        j = i
        while j > 0 and niz[j - 1] > trenutni:
            niz[j] = niz[j - 1]
            j = j - 1
        niz[j] = trenutni
    return niz


print(redjajUmetanjem([54, 26, 93, 17, 77, 31, 44, 55, 20]))

File: 20-sortiranje/redjanje-umetanjem.js
// https://skolakoda.org/redjanje-umetanjem
const slucajniNiz = require('../utils/slucajniNiz')

function redjajUmetanjem(niz) {
  for (let i = 1; i < niz.length; i++) {
    let j
    for (j = i; j > 0 && niz[j-1] > niz[i]; j--)
      niz[j] = niz[j-1]
    niz[j] = niz[i]
  }
  return niz
}

console.log(redjajUmetanjem(slucajniNiz(20)))
